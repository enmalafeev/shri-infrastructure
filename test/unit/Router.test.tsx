import React from 'react';
import { render, screen } from '@testing-library/react';
import { Router } from 'react-router';

import { Provider } from 'react-redux';
import { initStore } from '../../src/client/store';
import { createMemoryHistory } from 'history';
import { Application } from '../../src/client/Application';
import { ExampleApi, CartApi } from '../../src/client/api';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
 
describe('Проверка работы роутера', () => {
  it('Работает переход на страницу Catalog', () => {
    const history = createMemoryHistory({
      initialEntries: ['/hw/store/catalog'],
      initialIndex: 0
    });
    const basename = '/hw/store';
    const api = new ExampleApi(basename);
    const cart = new CartApi();
  
    const store = initStore(api, cart);
  
    const application = (
      <Router history={history}>
        <Provider store={store}>
            <Application />
        </Provider>
      </Router>
    );
    
    const { getByRole } = render(application);
    userEvent.click(getByRole('link', { name: /catalog/i }));
    const header = getByRole('heading', { name: /catalog/i });

    expect(header).toBeInTheDocument();
  });

  it('Работает переход на страницу Delivery', () => {
    const history = createMemoryHistory({
      initialEntries: ['/hw/store/delivery'],
      initialIndex: 0
    });
    const basename = '/hw/store';
    const api = new ExampleApi(basename);
    const cart = new CartApi();
  
    const store = initStore(api, cart);
  
    const application = (
      <Router history={history}>
        <Provider store={store}>
          <Application />
        </Provider>
      </Router>
    );
    
    const { getByRole } = render(application);
    userEvent.click(getByRole('link', { name: /delivery/i }));
    const header = getByRole('heading', { name: /delivery/i });
    const deliverySnapshot = renderer.create(application).toJSON();

    expect(header).toBeInTheDocument();
    expect(deliverySnapshot).toMatchSnapshot();
  });

  it('Работает переход на страницу Contacts', () => {
    const history = createMemoryHistory({
      initialEntries: ['/hw/store/contacts'],
      initialIndex: 0
    });
    const basename = '/hw/store';
    const api = new ExampleApi(basename);
    const cart = new CartApi();
  
    const store = initStore(api, cart);
  
    const application = (
      <Router history={history}>
        <Provider store={store}>
            <Application />
        </Provider>
      </Router>
    );
    
    const { getByRole } = render(application);
    userEvent.click(getByRole('link', { name: /contacts/i }));
    const header = getByRole('heading', { name: /contacts/i });
    const contactsSnapshot = renderer.create(application).toJSON();

    expect(header).toBeInTheDocument();
    expect(contactsSnapshot).toMatchSnapshot();
  });

  it('Работает переход на страницу Cart', () => {
    const history = createMemoryHistory({
      initialEntries: ['/hw/store/cart'],
      initialIndex: 0
    });
    const basename = '/hw/store';
    const api = new ExampleApi(basename);
    const cart = new CartApi();
  
    const store = initStore(api, cart);
  
    const application = (
      <Router history={history}>
        <Provider store={store}>
            <Application />
        </Provider>
      </Router>
    );
    
    const { getByRole } = render(application);
    userEvent.click(getByRole('link', { name: /cart/i }));
    const header = getByRole('heading', { name: /shopping cart/i })

    expect(header).toBeInTheDocument();
  });
})
 
 