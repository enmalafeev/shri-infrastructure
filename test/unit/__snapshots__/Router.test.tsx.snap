// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка работы роутера Работает переход на страницу Contacts 1`] = `
<div
  className="Application"
>
  <nav
    className="navbar navbar-expand-sm navbar-light bg-light"
  >
    <div
      className="container"
    >
      <a
        className="Application-Brand navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Example store
      </a>
      <button
        aria-label="Toggle navigation"
        className="Application-Toggler navbar-toggler"
        onClick={[Function]}
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="Application-Menu collapse navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/catalog"
            onClick={[Function]}
          >
            Catalog
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/delivery"
            onClick={[Function]}
          >
            Delivery
          </a>
          <a
            aria-current="page"
            className="nav-link active"
            href="/contacts"
            onClick={[Function]}
            style={Object {}}
          >
            Contacts
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/cart"
            onClick={[Function]}
          >
            Cart
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container pt-4"
  >
    <div
      className="Contacts"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h1>
            Contacts
          </h1>
          <p>
            Ut non consequatur aperiam ex dolores. Voluptatum harum consequatur est totam. Aut voluptatum aliquid aut optio et ea. Quaerat et eligendi minus quasi. Culpa voluptatem voluptatem dolores molestiae aut quos iure. Repellat aperiam ut aliquam iure. Veritatis magnam quisquam et dolorum recusandae aut.
          </p>
          <p>
            Molestias inventore illum architecto placeat molestias ipsam facilis ab quo. Rem dolore cum qui est reprehenderit assumenda voluptatem nisi ipsa. Unde libero quidem. Excepturi maiores vel quia. Neque facilis nobis minus veniam id. Eum cum eveniet accusantium molestias voluptas aut totam laborum aut. Ea molestiae ullam et. Quis ea ipsa culpa eligendi ab sit ea error suscipit. Quia ea ut minus distinctio quam eveniet nihil. Aut voluptate numquam ipsa dolorem et quas nemo.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Проверка работы роутера Работает переход на страницу Delivery 1`] = `
<div
  className="Application"
>
  <nav
    className="navbar navbar-expand-sm navbar-light bg-light"
  >
    <div
      className="container"
    >
      <a
        className="Application-Brand navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Example store
      </a>
      <button
        aria-label="Toggle navigation"
        className="Application-Toggler navbar-toggler"
        onClick={[Function]}
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="Application-Menu collapse navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/catalog"
            onClick={[Function]}
          >
            Catalog
          </a>
          <a
            aria-current="page"
            className="nav-link active"
            href="/delivery"
            onClick={[Function]}
            style={Object {}}
          >
            Delivery
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/contacts"
            onClick={[Function]}
          >
            Contacts
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/cart"
            onClick={[Function]}
          >
            Cart
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container pt-4"
  >
    <div
      className="Delivery"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h1>
            Delivery
          </h1>
          <p>
            Deserunt occaecati tempora. Qui occaecati est aliquam. Enim qui nulla ipsam. Incidunt impedit enim consequuntur amet at consequuntur vero. Dolor et ad facere asperiores iste est praesentium quaerat iure. Quibusdam mollitia autem quos voluptas quia est doloremque corporis et. Sed fuga quasi esse perspiciatis fugit maxime. Qui quidem amet.
          </p>
          <img
            className="Image w-25 mb-4"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
          />
          <p>
            Dolores magnam consequatur iste aliquam qui sint non ab. Culpa saepe omnis. Recusandae vel aperiam voluptates harum. Perspiciatis qui molestiae qui tempora quisquam. Mollitia voluptatum minus laboriosam. Dolor maiores possimus repudiandae praesentium hic eos. Veritatis et repellat.
          </p>
          <p>
            Pariatur nisi nobis hic ut facilis sunt rerum id error. Soluta nihil quisquam quia rerum illo. Ipsam et suscipit est iure incidunt quasi et eum. Culpa libero dignissimos recusandae. In magni sapiente non voluptas molestias. Deserunt quos quo placeat sunt. Ea necessitatibus dolores eaque ex aperiam sunt eius. Saepe aperiam aut. Quaerat natus consequatur aut est id saepe et aut facilis.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
